### PR TITLE
filelogger: make it idempotent

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/4481.

### CODE REVIEW

File: pkg/internal/filelogger/filelogger.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+			return microerror.Mask(err)
+		}
+
+		f.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("created log stream for pod %#q in namespace %#q", name, namespace))
+	}
+
+	{
+		f.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("starting logging job for pod %#q in namespace %#q", name, namespace))
+
+		go f.scan(ctx, logStream, logFilePath)
```
Comment (xh3b4sd) [inline pkg/internal/filelogger/filelogger.go]: When will this goroutine actually be terminated?

Comment (kopiczko) [reply pkg/internal/filelogger/filelogger.go]: When the `io.Copy` returns. Which is when the logged pod is deleted.

Comment (kopiczko) [reply pkg/internal/filelogger/filelogger.go]: This was implemented like that so far.

**Review by xh3b4sd — APPROVED** [review]
Alright. Makes sense.

**Pull request merged**